webapp/channels/src/components/advanced_text_editor/advanced_text_editor.tsx

```diff
@@ -457,14 +457,15 @@ const AdvanceTextEditor = ({
                         role='application'
                         id='advancedTextEditorCell'
                         data-a11y-sort-order='2'
-                        aria-label={Utils.localizeMessage(
-                            'channelView.login.successfull',
-                            'Login Successfull',
-                        ) + ' ' + ariaLabelMessageInput}
+                        aria-describedby="advancedTextEditorCellAriaLabel"
                         tabIndex={-1}
                         className='AdvancedTextEditor__cell a11y__region'
                     >
                         {labels}
+                        <label id="advancedTextEditorCellAriaLabel" style={{ display: "none", visibility: "hidden", height: 0 }}>{Utils.localizeMessage(
+                            'channelView.login.successfull',
+                            'Login Successfull',
+                        ) + ' ' + ariaLabelMessageInput}</label>
```

Comment:
I am not sure if this is exactly how this issue is supposed to be solved, honestly, I don't have the solution either. Please note that the issue is currently doesn't appear to be in Mattermost run in either Chrome, Firefox, but in Desktop app only.

Reply:
@M-ZubairAhmed After testing the changes I made to Mattermost on MacOS, it appears that the problem has been successfully resolved. I have attached screenshots for reference. The reason behind this solution is that when aria-label is added, Grammarly treats the textarea as a string, making it unable to detect. To address this issue, I included a label to prevent the textarea from being treated as a string, ensuring both accessibility is maintained and Grammarly detects it correctly. I conducted comprehensive testing using VoiceOver and Grammarly version 1.30.2.0 to validate the improvements


**
| <img width="971" alt="before_desktop" src="https://github.com/mattermost/mattermost/assets/62356575/e412dc9b-767b-4511-b783-333faba53a16"> | <img width="980" alt="after_desktop" src="https://github.com/mattermost/mattermost/assets/62356575/7f6bdb90-a66e-4082-aaea-aaffb2c6e647">  |**